### PR TITLE
Fix checks

### LINKED ISSUE #706: Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"
### Problem

In nonupdated browsers, not properly providing the proper `rel` attribute potentially allows the destination site to potentially hijack the browser via rogue JavaScript execution.

### Afflictions

```bash
C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\blog\labels.svelte
  18:7  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\button\button.svelte
  6:1  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\routes\(site)\blog\[id]\+page.svelte
  52:21  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\routes\(site)\quiz\more-info.svelte
  44:13  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer
```

### References

- <https://stackoverflow.com/a/50709724>

### LINKED ISSUE #707: A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event
### What happened?

Accessibility standards decree that visible, non-interactive elements with an on:click event must be accompanied by an `on:keydown`, `on:keyup`, or `on:keypress` event.

### Affected code

```bash
C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\nav\bar.svelte
  30:3  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\select\select.svelte
  21:5  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events
  25:9  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\routes\(site)\quiz\quiz.svelte
  143:7  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events
  158:9  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events
```

### PR DESCRIPTION
**Pull request by jaasonw**
fix these errors on every pr

image

noreferrer checks and keypress checks split into separate commits for easy cherrypicking

fix #706 and 707

### CODE REVIEW

File: src/lib/components/nav/bar.svelte [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
   <div class="blurrybar" />
 
-  <div class="backdrop" on:click={handleClose} />
+  <div class="backdrop" on:click={handleClose} on:keypress={handleClose} />
```
Comment (EthanThatOneKid) [inline src/lib/components/nav/bar.svelte]: Does this make the handler fire twice?

File: src/lib/components/select/select.svelte [hunk 2]
```diff
@@ -18,11 +18,17 @@
 
 <div class="term" name="school-year">
   <div class="option-box" class:active>
-    <div class="selected" on:click={toggleDropdown}>{currentValue}</div>
+    <div class="selected" on:click={toggleDropdown} on:keypress={toggleDropdown}>
```
Comment (EthanThatOneKid) [inline src/lib/components/select/select.svelte]: Does this make the handler fire twice?

File: src/lib/components/select/select.svelte [hunk 3]
```diff
@@ -18,11 +18,17 @@
...
+      {currentValue}
+    </div>
 
     <div class="option">
       {#each options as optionValue (optionValue)}
-        <div class="option-choice" on:click={() => handleOption(optionValue)}>
+        <div
+          class="option-choice"
+          on:click={() => handleOption(optionValue)}
+          on:keypress={() => handleOption(optionValue)}
```
Comment (EthanThatOneKid) [inline src/lib/components/select/select.svelte]: Does this make the handler fire twice?

File: src/routes/(site)/quiz/quiz.svelte [hunk 4]
```diff
@@ -143,6 +143,7 @@
       <div
         class="result-grid-box"
         on:click={() => showTeamDetails(TEAMS[match])}
+        on:keypress={() => showTeamDetails(TEAMS[match])}
```
Comment (EthanThatOneKid) [inline src/routes/(site)/quiz/quiz.svelte]: Does this make the handler fire twice?

File: src/routes/(site)/quiz/quiz.svelte [hunk 5]
```diff
@@ -159,6 +160,7 @@
           class="result-grid-box"
           style={`--border-color: ${team.color}`}
           on:click={() => showTeamDetails(team)}
+          on:keypress={() => showTeamDetails(team)}
```
Comment (EthanThatOneKid) [inline src/routes/(site)/quiz/quiz.svelte]: Does this make the handler fire twice?

**Review by EthanThatOneKid — COMMENTED** [review]
Issues #706 and #707 were intended to be resolved in separate PRs but since these changes seem to be relatively small, I suppose we can go ahead and squash these together?

File: src/lib/components/select/select.svelte (hunk 2 above)
Comment (jaasonw) [reply src/lib/components/select/select.svelte]: no

File: src/lib/components/select/select.svelte (hunk 3 above)
Comment (jaasonw) [reply src/lib/components/select/select.svelte]: no

File: src/routes/(site)/quiz/quiz.svelte (hunk 4 above)
Comment (jaasonw) [reply src/routes/(site)/quiz/quiz.svelte]: no

File: src/routes/(site)/quiz/quiz.svelte (hunk 5 above)
Comment (jaasonw) [reply src/routes/(site)/quiz/quiz.svelte]: no

File: src/lib/components/nav/bar.svelte (hunk 1 above)
Comment (jaasonw) [reply src/lib/components/nav/bar.svelte]: no

**Comment by jaasonw** [comment]
didnt realize there was separate issue for both

**Pull request closed without merging**